docs/app/documentation/component-docs/menu/examples/menu-keyboard-support-example.component.ts

```diff
@@ -0,0 +1,38 @@
+import { AfterViewInit, Component, QueryList, ViewChildren } from '@angular/core';
+import { MenuKeyboardService } from '../../../../../../library/src/lib/menu/menu-keyboard.service';
+import { MenuItemDirective } from '../../../../../../library/src/lib/menu/menu-item.directive';
+
+@Component({
+    selector: 'fd-menu-keyboard-support-example',
+    templateUrl: 'menu-keyboard-support-example.component.html',
+    providers: [
+        MenuKeyboardService
+    ]
+})
+export class MenuKeyboardSupportExampleComponent implements AfterViewInit {
+
+    @ViewChildren(MenuItemDirective)
+    menuItems: QueryList<MenuItemDirective>;
+
+    constructor (private menuKeyboardService: MenuKeyboardService) {}
+
+    public focusFirst() {
+        this.menuItems.first.focus();
+    }
+
+    public escapeAfterListFunction = () => {
+        alert('Escaped After list');
+    };
+
+    public onClick() {
+        alert('Element clicked');
+    }
+
+    public handleKeyDown(event: KeyboardEvent, index: number): void {
+        this.menuKeyboardService.keyDownHandler(event, index, this.menuItems.toArray());
+    }
+
+    ngAfterViewInit(): void {
+        this.menuKeyboardService.focusEscapeAfterList = this.escapeAfterListFunction;
```

Comment:
Not sure I like this approach to the keyboard navigation, it's a bit confusing for developers to use I think. Maybe we're better off without it completely and just implementing it for individual components, I'm not sure.

Reply:
I'm sure these functions would be easy for developers to use. It's completely optional, so devs can handle events by themselves. I would rather keep it, this approach will greatly reduce amount of code in individual components.

Reply:
@MattL75